### ISSUE TITLE
Unable to perform camera position optimisation in spherical coordinates

### ISSUE THREAD

**Issue by thechargedneutron** [facebookresearch/pytorch3d#379]
The camera position optimisation tutorial given [here](https://github.com/facebookresearch/pytorch3d/blob/master/docs/tutorials/camera_position_optimization_with_differentiable_rendering.ipynb) works pretty well in training cartesian coordinates. However, the same does not train well with spherical coordinates. Scaling the angles to [0, 1] and changing the learning rate does not seem to work.

Changes to the code
-----------------------

Changed
`self.camera_position = nn.Parameter(
            torch.from_numpy(np.array([3.0,  6.9, +2.5], dtype=np.float32)).to(meshes.device))`
to
`self.radius = 3.0          #Same as reference radius`
`self.camera_angles = nn.Parameter(
            torch.from_numpy(np.array([40.0,  80.0], dtype=np.float32)).to(meshes.device))`

and the rendering from
`R = look_at_rotation(self.camera_position[None, :], device=self.device)  # (1, 3, 3)`
`T = -torch.bmm(R.transpose(1, 2), self.camera_position[None, :, None])[:, :, 0]   # (1, 3)`
to
`R, T = look_at_view_transform(self.radius, self.camera_angles[0], self.camera_angles[1], device=self.device)`

Also, modified the learning rate to a larger value since the range of angles is more than the variation in cartesian coordinates.

Any leads on how to modify the training procedure will be highly appreciated. Thanks for this awesome library.

**Comment by bottler** [comment]
You're solving a different optimization problem to the original. In the original, the camera can move in 3D. It moves towards the teapot to get a bigger view which matches the reference. In your version you fix the radius (it is a python float not a parameter of the model), so the camera is constrained to a sphere, so the size of the rendered teapot is roughly constant. What are you hoping the optimization will lead to in your fixed case?

**Comment by thechargedneutron** [comment]
Yes, the radius is deliberately kept constant. Basically, I want to "learn" given a target image, from which azimuth and elevation should I look so that the teapot is as same as possible, i.e. the error is the least. 
Naturally, it is not un unconstrainted 3D movement, but the movement is restricted to a sphere only, i.e. a 2D motion.
Does this clarify my objective?

**Comment by nikhilaravi** [comment]
@thechargedneutron are you also generating the target image using a similar set up? e.g. fixed radius and randomly selected azim/elevation?

**Comment by thechargedneutron** [comment]
Yes. The location is also the same (radius 3, elevation 50 and azimuth 0) as the tutorial.

`distance = 3   # distance from camera to the object`
`elevation = 50.0   # angle of elevation in degrees`
`azimuth = 0.0  # No rotation so the camera is positioned on the +Z axis. `

**Comment by gkioxari** [comment]
Hi @thechargedneutron! 

It's not clear from your issue whether this is a PyTorch3D bug. 

The github issues for this codebase should be used for bug reports in the codebase or feature enhancement requests. If you believe this is a bug, please report back with a compact code snippet, the data and what you were expecting the output of the function to be and what you actually get. This will allow us to reproduce and fix the issue. 

From the description of your issue, this doesn't seem to be a bug but a research question of how to do X. We wish we could help you with your question. Unfortunately, we don't have the capacity to do this, otherwise that is all we would do. I do encourage you though once you figure this out, to report back as other users might find it useful in their question to perform the same task!

**Comment by gkioxari** [comment]
Closing this for now! Please comment back if you ever figure out how to solve your optimization question.
